### DISCUSSION TITLE
Finding the main subject of a sentence

### DISCUSSION THREAD

**Discussion by mattkallo** [explosion/spaCy#12943] [Help: Coding & Implementations] [answered]
I have the below sentence, I am trying to find the main subject/entity of the sentence. Is there a way to find it? When I do the dependency parsing/pos tagging, it doesn't give nsubj. 

```
doc = nlp("Dr. Ben Gupta, the chief medical officer of Amazon Pharmacy") 
for chunk in doc.noun_chunks:
    print(chunk.text + " => " + chunk.root.text + " => " +  chunk.root.dep_ + " => " + 
            chunk.root.head.text)

Dr. Ben Gupta => Gupta => ROOT => Gupta
the chief medical officer => officer => appos => Gupta
Amazon Pharmacy => Pharmacy => pobj => of
```

I am trying to extract the main subject - "Dr. Ben Gupta"

**Comment by svlandeg** [comment, marked as answer]
Hi @mattkallo,

I think I understand what you're trying to do, but you'll need to allow for some syntactic variation across sentences to be able to do this. For instance, in the sentence that you used, there is no actual verb, and so there also isn't a "nsubj" grammar relation. Instead, as you've found, "Gupta" is the ROOT word of the sentence.

Now, suppose we'd have the sentence
```
"Dr. Ben Gupta, the chief medical officer of Amazon Pharmacy, is happy."
```
then your code snippet would in fact show:
```
Dr. Ben Gupta => Gupta => nsubj => is
```
thus marking "Dr Ben Gupta" as the main "entity", if you like.

On a related topic, which might be relevant for what you're trying to do, have you had a look at our built-in [`DependencyMatcher`](https://spacy.io/usage/rule-based-matching#dependencymatcher)?

**Reply by mattkallo** [reply]
Thanks @svlandeg 

The example I gave was automatically picked up in the coreference cluster by the model,  don't have an option to change the sentence/add more context. See the coreference cluster below.  I am trying to find a generalized solution that works in other scenarios as well. 

```
["Dr. Ben Gupta, the chief medical officer of Amazon Pharmacy,", "he"]
```
Other similar situations

```
["Amazon Pharmacy, a subsidiary of Amazon,", "It"]
["Sister of John Doe, Jane Doe", "she"]

```
I am thinking whether we can pick the ROOT as the right entity if ROOT is a noun instead of verb.  Do you see any issue with that approach?

**Reply by svlandeg** [reply]
It really depends on your use-case - it's difficult for us to make generic recommendations on what you should do. But yes, some kind of fallback if-else approach, and/or a combination with an NER model and/or DependencyMatcher could work.
